liota/entities/devices/sensor_tag.py

```diff
@@ -0,0 +1,451 @@
+        self._acce_queue = None
+        self._magn_queue = None
+        self._gyro_queue = None
+        self._bat_level_queue = None
+        self._light_queue = None
+        if not isinstance(sensors, list) or len(sensors) == 0:
+            log.error("List of sensors to be enabled is expected.")
+            raise TypeError("List of sensors to be enabled is expected.")
+        # List of sensors to be enabled
+        self._enable_sensors_list = sensors
+        # Connects with re-try mechanism
+        self._re_connect()
+        self.start()
+
+    def get_sensor_tag(self):
+        """
+        Returns SensorTagDevice entity.
+        :return: SensorTagDevice entity object.
+        """
+        return self.tag
+
+    def get_temperature(self):
+        """
+        Blocks until value is available in queue
+        :return: (ambient_temp, target_temp) tuple in degC
+        """
+        return self._temp_queue.get(True, self._sampling_interval_sec+1)
+
+    def get_humidity(self):
+        """
+        Blocks until (sampling_interval_sec + 1 sec)
+        :return: (ambient_temp, rel_humidity) tuple
+        """
+        return self._humi_queue.get(True, self._sampling_interval_sec+1)
+
+    def get_barometer(self):
+        """
+        Blocks until (sampling_interval_sec + 1 sec)
+        :return: (ambient_temp in degC, pressure in millibars) tuple
+        """
+        return self._baro_queue.get(True, self._sampling_interval_sec+1)
+
+    def get_accelerometer(self):
+        """
+        Blocks until (sampling_interval_sec + 1 sec)
+        :return: (x_accel, y_accel, z_accel) tuple in units of g
+        """
+        return self._acce_queue.get(True, self._sampling_interval_sec+1)
+
+    def get_magnetometer(self):
+        """
+        Blocks until (sampling_interval_sec + 1 sec)
+        :return: (x_mag, y_mag, z_mag) in units of uT
+        """
+        return self._magn_queue.get(True, self._sampling_interval_sec+1)
+
+    def get_gyroscope(self):
+        """
+        Blocks until (sampling_interval_sec + 1 sec)
+        :return: (x_gyro, y_gyro, z_gyro) tple in units of degrees/sec
+        """
+        return self._gyro_queue.get(True, self._sampling_interval_sec+1)
+
+    def get_battery_level(self):
+        """
+        Blocks until (sampling_interval_sec + 1 sec)
+        :return: battery level in percent
+        """
+        return self._bat_level_queue.get(True, self._sampling_interval_sec+1)
+
+    def get_light_level(self):
+        """
+        Blocks until (sampling_interval_sec + 1 sec)
+        :return: value in lux
+        """
+        return self._light_queue.get(True, self._sampling_interval_sec+1)
+
+    def _connect(self):
+        """
+        Connects with a SensorTag Device and enables the specified sensors.
+        :return: None
+        """
+        log.info("Connecting to SensorTag Device: {0} with MAC_ADDRESS: {1}".format(self.device_name, self.device_mac))
+        self.tag = SensorTagDevice(
```

Comment:
This line should be part of user code(Initializing Device class) shouldn't be in the core Liota.